File: week_1/day-3-two-pointer/remove_duplicates_sorted_array.py
nums = [1, 1, 2, 2, 3, 3, 4]

def removeDuplicates(nums):
    if not nums:
        return 0

    slow = 0
    for fast in range(1, len(nums)):
        if nums[fast] != nums[slow]:
            slow += 1
            nums[slow] = nums[fast]

    return slow + 1

length = removeDuplicates(nums)

print(length)          # Output: 4
print(nums[:length])   # Output: [1, 2, 3, 4]

File: week_1/day-3-two-pointer/remove_duplicates_sorted_array.js
const nums = [1, 1, 2, 2, 3, 3, 4];

function removeDuplicates(nums) {
    if (nums.length === 0) return 0;

    let slow = 0;
    for (let fast = 1; fast < nums.length; fast++) {
        if (nums[fast] !== nums[slow]) {
            slow++;
            nums[slow] = nums[fast];
        }
    }
    return slow + 1;
}

const length = removeDuplicates(nums);

console.log(length);                // Output: 4
console.log(nums.slice(0, length)); // Output: [1, 2, 3, 4]
